### ISSUE TITLE
cadvisor 0.26.0 /metrics is not work

### ISSUE THREAD

**Issue by pangzheng** [google/cadvisor#1671]
src build file

```
FROM golang:latest
MAINTAINER vmarmol@google.com

RUN apt-get install -y git dmsetup
RUN git clone https://github.com/google/cadvisor.git /go/src/github.com/google/cadvisor
RUN cd /go/src/github.com/google/cadvisor && make

ENTRYPOINT ["/go/src/github.com/google/cadvisor/cadvisor"]
```

build work image

```
FROM alpine:3.4
MAINTAINER dengnan@google.com vmarmol@google.com vishnuk@google.com jimmidyson@gmail.com stclair@google.com

ENV GLIBC_VERSION "2.23-r3"

RUN apk --no-cache add ca-certificates wget device-mapper && \
    apk --no-cache add zfs --repository http://dl-3.alpinelinux.org/alpine/edge/main/ && \
    apk --no-cache add thin-provisioning-tools --repository http://dl-3.alpinelinux.org/alpine/edge/main/ && \     <-- i change there，The original does not work
    wget -q -O /etc/apk/keys/sgerrand.rsa.pub https://raw.githubusercontent.com/sgerrand/alpine-pkg-glibc/master/sgerrand.rsa.pub && \
    wget https://github.com/sgerrand/alpine-pkg-glibc/releases/download/${GLIBC_VERSION}/glibc-${GLIBC_VERSION}.apk && \
    wget https://github.com/andyshinn/alpine-pkg-glibc/releases/download/${GLIBC_VERSION}/glibc-bin-${GLIBC_VERSION}.apk && \
    apk add glibc-${GLIBC_VERSION}.apk glibc-bin-${GLIBC_VERSION}.apk && \
    /usr/glibc-compat/sbin/ldconfig /lib /usr/glibc-compat/lib && \
    echo 'hosts: files mdns4_minimal [NOTFOUND=return] dns mdns4' >> /etc/nsswitch.conf && \
    rm -rf /var/cache/apk/*

# Grab cadvisor from the staging directory.
ADD cadvisor /usr/bin/cadvisor

EXPOSE 8080
ENTRYPOINT ["/usr/bin/cadvisor", "-logtostderr"]
```

error info

```
An error has occurred during metrics gathering:

2293 error(s) occurred:
* collected metric container_start_time_seconds label:<name:"container_label_build_date" value:"2016-03-31" > label:<name:"container_label_license" value:"GPLv2" > label:<name:"container_label_name" value:"CentOS Base Image" > label:<name:"container_label_vendor" value:"CentOS" > label:<name:"id" value:"/docker/3dd5e11f25fd2431432bd12c7b8ae44ccfbd3931cba87e6c55f52d5d0785ba46" > label:<name:"image" value:"centos7-mesos-1.2.0-0.2-master" > label:<name:"name" value:"mesos-master" > gauge:<value:1.49689091e+09 >  has label dimensions inconsistent with previously collected metrics in the same metric family
````

**Comment by pangzheng** [comment]
Please help answer，thx

**Comment by rickardrosen** [comment]
I'm also seeing this.

**Comment by hajhatten** [comment]
Did get womething similar as @pangzheng at work, where we use the --device flag in docker, to get /dev/zfs metrics from cadvisor.

Tried google/cadvisor-canary out on my freebsd machine at home, and it works perfectly for 40s, then prints out the following:

```
2 error(s) occurred:
* collected metric container_fs_reads_total label:<name:"device" value:"/dev/loop0" > label:<name:"id" value:"/" > counter:<value:0 >  was collected before with the same name and label values
* collected metric container_fs_writes_total label:<name:"device" value:"/dev/loop0" > label:<name:"id" value:"/" > counter:<value:0 >  was collected before with the same name and label values
```

Cannot recreate this in google/cadvisor-canary or with a custom build with the release-v0.26 branch on my docker toolbox machine. I'm not running the --device flag there.

**Comment by dashpole** [comment]
fixed by #1679, and in the 0.26.1 release.
